smart-contracts/src/contracts/TrashToken.sol

```diff
@@ -0,0 +1,86 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.4;
+
+import "@openzeppelin/contracts-upgradeable/token/ERC20/presets/ERC20PresetMinterPauserUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC20/extensions/ERC20VotesUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
+
+struct TokenHolder {
+  address account;
+  uint amount;
+}
+
+contract TrashToken is Initializable, ERC20VotesUpgradeable, UUPSUpgradeable, ERC20PresetMinterPauserUpgradeable {
+  address public ratContract;
+  address public closetContract;
+  bytes32 private constant ADMIN_ROLE = keccak256("ADMIN_ROLE");
+  TokenHolder[] public tokenHolders;
```

Comment:
Instead of an array, we should just map of an address to the amount they own.

Reply:
I remember why I did it this way.

Solidity doesn't allow you to return arbitrary mappings from functions without passing a key. So if we want to list all token holders we have to put them in an array.